libs/application/templates/children-residence-change/src/fields/SelectChildren/index.tsx

```diff
@@ -1,18 +1,45 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { useIntl } from 'react-intl'
 import { CheckboxController } from '@island.is/shared/form-fields'
 import { Box, Text } from '@island.is/island-ui/core'
 import { selectChildren } from '../../lib/messages'
-import { CRCFieldBaseProps } from '../../types'
+import { CRCFieldBaseProps, Child } from '../../types'
 import { DescriptionText } from '../components'
 
+const shouldBeDisabled = (
+  children: Child[],
+  selectedChildren: string[],
+  childOption: Child,
+) => {
+  if (!selectedChildren || selectedChildren?.length === 0) {
+    return false
+  }
+  const firstSelectedChild = children.find(
+    (child) => selectedChildren[0] === child.nationalId,
+  )
+  if (
+    firstSelectedChild?.livesWithApplicant !== childOption.livesWithApplicant ||
+    firstSelectedChild?.otherParent.nationalId !==
+      childOption.otherParent.nationalId
+  ) {
+    return true
+  }
+  return false
+}
+
 const SelectChildren = ({ field, application, error }: CRCFieldBaseProps) => {
   const { id, disabled } = field
   const { formatMessage } = useIntl()
   const {
     externalData: { nationalRegistry },
+    answers,
   } = application
   const children = nationalRegistry.data.children
+  const currentAnswer = answers.selectedChildren
+  const [selectedChildrenState, setSelectedChildrenState] = useState<string[]>(
+    currentAnswer,
+  )
```

Comment:
Since `currentAnswer` can be undefined isn't the type here `string[] | undefined` ?